### DISCUSSION TITLE
How to fix ApiPlatform User deprecated message ?

### DISCUSSION THREAD

**Discussion by Asmitta-01** [symfony/symfony#47251] [Q&A] [answered]
I am working with ApiPlatform and i created two entities User and Event. When i update for example th Event entity i get this message even 10 times in the console:

>2022-08-11T09:46:27+02:00 [info] User Deprecated: The "ApiPlatform\Core\Swagger\Serializer\ApiGatewayNormalizer::supportsNormalization()" method will require a new "array $context" argument in the next major version of its interface "Symfony\Component\Serializer\Normalizer\NormalizerInterface", not defining it is deprecated.

And at the end:
>2022-08-11T09:46:27+02:00 [info] User Deprecated: The "Symfony\Bridge\Doctrine\Logger\DbalLogger" class implements "Doctrine\DBAL\Logging\SQLLogger" that 
is deprecated Use {@see \Doctrine\DBAL\Logging\Middleware} or implement {@see \Doctrine\DBAL\Driver\Middleware} instead.

How can i fix it please?

**Comment by MatTheCat** [comment, marked as answer]
This is not a bug but a deprecation. You generally don’t need to do anything when it comes from vendor code, aside upgrading them.

The first one says that you won’t be able to upgrade to Symfony ≥ 6.1 without upgrading api-platform/core to version 3 first.

The second tells you you use doctrine/dbal ≥ 3.2 but some of your vendors still use its deprecated `Doctrine\DBAL\Logging\SQLLogger` class.

**Reply by Asmitta-01** [reply]
Okay thanks
But how can i hide it ? seeing these message each time is not really pleasan

**Reply by MatTheCat** [reply]
If you’re using the monolog bundle you should be able to silence the `deprecation` channel. Never did it though so cannot help further.
